raft/src/main/java/org/apache/kafka/snapshot/SnapshotWriter.java

```diff
@@ -78,6 +83,76 @@ public SnapshotWriter(
         );
     }
 
+    /**
+     * Adds a {@link MetadataSnapshotHeaderRecord} to snapshot
+     *
+     * @throws IllegalStateException if the snapshot is not empty
+     */
+    private void initializeSnapshotWithHeader() {
+        if (snapshot.sizeInBytes() != 0) {
+            throw new IllegalStateException("Initializing new snapshot (ID: "
+                    + snapshot.snapshotId().epoch + ", " + snapshot.snapshotId().offset
+                    + ") with a non-empty file");
```

Comment:
You can rely on `OffsetAndEpoch::toString`. For example, `String.format("Initializing with non-empty snapshot  (%s)", snapshot.snapshotId())`.

Changed the message because it is possible that the `RawSnapshotWriter` is not a file but an in-memory snapshot.

This comment always applies to the footer message.

Reply:
Will do.